patches/third_party-blink-renderer-core-fileapi-public_url_manager.cc.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/third_party/blink/renderer/core/fileapi/public_url_manager.cc b/third_party/blink/renderer/core/fileapi/public_url_manager.cc
+index 17c4f34c63b71f6c83c188c1a9ab324b470308b6..4dcb50f260d7b0a22f6c299ca2b6d2f8e51b6d0d 100644
+--- a/third_party/blink/renderer/core/fileapi/public_url_manager.cc
++++ b/third_party/blink/renderer/core/fileapi/public_url_manager.cc
+@@ -62,7 +62,7 @@ PublicURLManager::PublicURLManager(ExecutionContext* context)
+       is_stopped_(false),
+       url_store_(context) {
+   BlobDataHandle::GetBlobRegistry()->URLStoreForOrigin(
+-      context->GetSecurityOrigin(),
```

Comment:
I think you can get rid of this patch with something like
```
#define URLStoreForOrigin(ORIGIN, URL_STORE) URLStoreForOrigin(!GetEphemeralOrOriginalSecurityOrigin(context).is_null() ? GetEphemeralOrOriginalSecurityOrigin(context) : ORIGIN, URL_STORE)
```

Reply:
actually why do we need this patch at all? This ends up calling `BlobURLStoreImpl` which we are overriding anyway

Reply:
I guess we don't have access to the content settings there, but according to @pes10k this doesn't even need to be dependent on content settings which could potentially simplify a lot of this.